src/MyEventEmitter.js

```diff
@@ -1,12 +1,79 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  on(event, callback) {
+    this.events[event] = this.events[event] || [];
+    this.events[event].push(callback);
+  }
+
+  once(event, callback) {
+    function onceCallback(data) {
+      callback();
+      this.off(event, onceCallback);
+    }
+
+    this.on(event, onceCallback);
+  }
+  off(event, callback) {
+    const listener = this.events[event] || [];
+
+    if (!listener) {
+      return this;
+    }
+
+    for (let i = this.length; i > 0; i++) {
+      if (listener[i] === callback) {
+        listener.splice(i, 1);
+      }
+    }
```

Comment:
In your solution listener is always present as you set it to an empty array if it's undefined
![image](https://github.com/mate-academy/node_custom-event-emitter/assets/63243136/745e3788-133f-4ec8-be99-d5aca139d22c)

Also, use `filter` instead of `fori` loop and `splice`

Something like this:
```javascript
if (this.events[event]) {
  this.events[event] = this.events[event].filter((listener) => listener !== callback);
}
```